pkg/apiserver/watch.go

```diff
@@ -181,11 +182,20 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	flusher.Flush()
 
+	delay := wait.NewChannelDelayedAction(
+		wait.Interval{Count: 0, Interval: 5 * time.Millisecond},
+		wait.Interval{Count: 10, Interval: 100 * time.Millisecond},
+		wait.Interval{Count: 100, Interval: 500 * time.Millisecond},
```

Comment:
Just initial experimentation, Im' going to refine this a bit today with a
ramp up for the first interval (we shouldn't start waiting until we detect
X counts in a window).

On Tue, Apr 26, 2016 at 7:24 AM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/apiserver/watch.go
> <https://github.com/kubernetes/kubernetes/pull/24768#discussion_r61068823>
> :
>
> > @@ -181,11 +182,20 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
> >  	w.WriteHeader(http.StatusOK)
> >  	flusher.Flush()
> >
> > +	delay := wait.NewChannelDelayedAction(
> > +		wait.Interval{Count: 0, Interval: 5 * time.Millisecond},
> > +		wait.Interval{Count: 10, Interval: 100 * time.Millisecond},
> > +		wait.Interval{Count: 100, Interval: 500 * time.Millisecond},
>
> I'm wondering where those values come from. Where do they come from?
>
> My personal feeling is that 500ms may be too high, but if those are based
> on some experiment, can you please describe it?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24768/files/91a0c160abd3ac01d0dbe9a89b3b5131c0cad419#r61068823>
>